Pod/Classes/WPMediaCollectionViewController.m

```diff
@@ -806,8 +806,12 @@ - (void)previewingContext:(id <UIViewControllerPreviewing>)previewingContext com
     [self.navigationController pushViewController:viewControllerToCommit animated:YES];
 }
 
-- (void)assetViewController:(WPAssetViewController *)assetPreviewVC selectionChange:(BOOL)selected
+#pragma mark - WPAssetViewControllerDelegate
+
+- (void)assetViewController:(WPAssetViewController *)assetPreviewVC selectionChanged:(BOOL)selected
 {
+    [self.navigationController popViewControllerAnimated:YES];
```

Comment:
Do you want to check that the asset view controller is on the navigation stack before calling pop?